common-healing.lic

```diff
@@ -162,13 +162,33 @@ module DRCH
   module_function
 
   def check_health
+    all_parasites = [
+      /(?:small|large) (?:black|red) blood mite/,
+      /red sand leech/,
+      /albino forest leech/,
+      /(?:green|red) blood worm/,
+      /Retch maggot/
+    ]
+    all_parasites_re = Regexp.union(all_parasites)
+    wounds_line = nil
+    parasites_line = nil
+    diseased = false
+    poisoned = false
+
     DRC.bput('health', 'You have')
+    pause 0.5
     data = reget(300).reverse
-    wounds_line = nil
     data.each do |line|
-      if line =~ /^You have/ && line !~ /^You have no significant injuries\./ && line !~ /^You have .* lodged .* into your/ && line !~ /^You have .* poisoned/
+      case line
+      when /^You have (?!no significant injuries)(?!.* lodged into your)(?!.* infection)(?!.* poisoned)(?!.* #{all_parasites_re})/
```

Comment:
I'm a little confused, why are the infection, poison, and parasite lines matched here and in their own `when`?

Reply:
Its a negative look ahead match on 'You have' but not followed by anything after `?!`